src/main/windows/mainWindow.js

```diff
@@ -52,6 +52,7 @@ function createMainWindow(config, options) {
     }
 
     const {maximized: windowIsMaximized} = windowOptions;
+    const fullscreen = process.platform === 'darwin' && windowIsMaximized && windowOptions.x === 0 && windowOptions.y === 0;
```

Comment:
since we save the info in the file, why not save the fullscreen state so we don't have to guess if it is fullscreen or maximized?